undefined reference to `__fprint_chk`

#### Expected behavior

go test command runs successfully when building with github.com/apache/pulsar-client-go@master on alpine

#### Actual behavior
go test fails with the following output:

`# github.com/valyala/gozstd
/usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: vendor/github.com/valyala/gozstd/libzstd_linux_amd64.a(zdict.o): in function `ZDICT_analyzeEntropy':
zdict.c:(.text+0x7ce): undefined reference to `__fprintf_chk'
/usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: vendor/github.com/valyala/gozstd/libzstd_linux_amd64.a(zdict.o): in function `ZDICT_analyzePos':
zdict.c:(.text+0x1832): undefined reference to `__fprintf_chk'
/usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: zdict.c:(.text+0x18c2): undefined reference to `__fprintf_chk'
/usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: vendor/github.com/valyala/gozstd/libzstd_linux_amd64.a(zdict.o): in function `ZDICT_finalizeDictionary':
zdict.c:(.text+0x1b24): undefined reference to `__fprintf_chk'
/usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: vendor/github.com/valyala/gozstd/libzstd_linux_amd64.a(zdict.o): in function `ZDICT_trainFromBuffer_unsafe_legacy':
zdict.c:(.text+0x1d54): undefined reference to `__fprintf_chk'
/usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: vendor/github.com/valyala/gozstd/libzstd_linux_amd64.a(zdict.o):zdict.c:(.text+0x1dd4): more undefined references to `__fprintf_chk' follow
/usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: vendor/github.com/valyala/gozstd/libzstd_linux_amd64.a(entropy_common.o): in function `FSE_readNCount':
entropy_common.c:(.text+0x345): undefined reference to `__memcpy_chk'
/usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: vendor/github.com/valyala/gozstd/libzstd_linux_amd64.a(fastcover.o): in function `FASTCOVER_buildDictionary.isra.6':
fastcover.c:(.text+0x3a9): undefined reference to `__fprintf_chk'
/usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: fastcover.c:(.text+0x44f): undefined reference to `__fprintf_chk'
/usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: fastcover.c:(.text+0x488): undefined reference to `__fprintf_chk'
/usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: vendor/github.com/valyala/gozstd/libzstd_linux_amd64.a(fastcover.o): in function `FASTCOVER_ctx_init':
fastcover.c:(.text+0x8a0): undefined reference to `__fprintf_chk'
/usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: fastcover.c:(.text+0x8e6): undefined reference to `__fprintf_chk'
/usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: vendor/github.com/valyala/gozstd/libzstd_linux_amd64.a(fastcover.o):fastcover.c:(.text+0xbc7): more undefined references to `__fprintf_chk' follow
collect2: error: ld returned 1 exit status
The command '/bin/sh -c go test -mod=vendor ./...' returned a non-zero code: 2`

#### Steps to reproduce

Create project with pulsar-client-go dependency
go get github.com/apache/pulsar-client-go/pulsar@master
go mod tidy
go mod vendor
run go test -mod=vendor ./... on alpine docker image (golang:alpine)


#### System configuration
**Pulsar version**: apachepulsar/pulsar v2.5.2

--- a/go.mod
+++ b/go.mod
@@ -3,6 +3,7 @@ module github.com/apache/pulsar-client-go
 go 1.12
 
 require (
+	github.com/DataDog/zstd v1.4.6-0.20200617134701-89f69fb7df32
 	github.com/beefsack/go-rate v0.0.0-20180408011153-efa7637bb9b6
 	github.com/bmizerany/perks v0.0.0-20141205001514-d9a9656a3a4b
 	github.com/golang/protobuf v1.3.1
@@ -15,6 +16,5 @@ require (
 	github.com/spf13/cobra v0.0.3
 	github.com/spf13/pflag v1.0.3 // indirect
 	github.com/stretchr/testify v1.4.0
-	github.com/valyala/gozstd v1.7.0
 	github.com/yahoo/athenz v1.8.55
 )

--- a/pulsar/internal/compression/zstd_cgo.go
+++ b/pulsar/internal/compression/zstd_cgo.go
@@ -24,29 +24,37 @@
 package compression
 
 import (
-	zstd "github.com/valyala/gozstd"
+	"github.com/DataDog/zstd"
+	log "github.com/sirupsen/logrus"
 )
 
 type zstdCGoProvider struct {
+	ctx              zstd.Ctx
 	compressionLevel int
 }
 
 func newCGoZStdProvider(compressionLevel int) Provider {
 	return &zstdCGoProvider{
 		compressionLevel: compressionLevel,
+		ctx:              zstd.NewCtx(),
 	}
 }
 
 func NewZStdProvider() Provider {
-	return newCGoZStdProvider(zstd.DefaultCompressionLevel)
+	return newCGoZStdProvider(zstd.DefaultCompression)
 }
 
 func (z *zstdCGoProvider) Compress(data []byte) []byte {
-	return zstd.CompressLevel(nil, data, z.compressionLevel)
+	out, err := z.ctx.CompressLevel(nil, data, z.compressionLevel)
+	if err != nil {
+		log.WithError(err).Fatal("Failed to compress")
+	}
+
+	return out
 }
 
 func (z *zstdCGoProvider) Decompress(compressedData []byte, originalSize int) ([]byte, error) {
-	return zstd.Decompress(nil, compressedData)
+	return z.ctx.Decompress(nil, compressedData)
 }
 
 func (z *zstdCGoProvider) Close() error {

